torch/csrc/jit/passes/quantization.cpp

```diff
@@ -692,6 +695,149 @@ c10::optional<IValue> toTwoElementIntList(Value* v) {
+    const script::Module& module,
+    const std::vector<std::string>& path) {
+  script::Module m = module;
+  for (size_t i = 0; i < path.size(); ++i) {
+    m = m.attr(path[i]).toModule();
+  }
+  return m;
+}
+
+std::string addChildModule(
+    script::Module module,
+    const script::Module& child_module,
+    const std::vector<std::string>& path) {
+  script::Module cur = module;
+  TORCH_INTERNAL_ASSERT(
+      path.size() > 0, "path must have at least one element.");
+  for (size_t i = 0; i < path.size() - 1; ++i) {
+    cur = module.attr(path[i]).toModule();
+  }
+
+  std::string cur_name = path[path.size() - 1];
+  int uid = 0;
+  std::string child_name = cur_name + "_" + c10::to_string(uid++);
+  while (module.hasattr(child_name)) {
+    child_name = cur_name + "_" + c10::to_string(uid++);
+  }
+  module.register_module(child_name, child_module.clone());
+  return child_name;
+}
+
+// Check if value is the input of the graph
+bool hitGraphInput(Value* value) {
+  Graph* graph = value->owningGraph();
+  const auto& inputs = graph->inputs();
+  return std::find(inputs.begin(), inputs.end(), value) != inputs.end();
+}
+
+Value* getModuleAccessPath(Value* instance, std::vector<std::string>& path) {
+  // Iterator to traverse back the GetAttr calls
+  Value* iter = instance;
+  // trace back the instance to recover the path of the submodule
+  while (!hitGraphInput(iter) && iter->node()->kind() == prim::GetAttr) {
+    Node* get_attr = iter->node();
+    // record the name of GetAttr
+    path.push_back(get_attr->s(attr::name));
+    // trace back the chain of GetAttr
+    iter = get_attr->inputs()[0];
+  }
+  return iter;
+}
+
+// A helper class to make uses of module unique
+class ModuleUseDeduper {
+ public:
+  ModuleUseDeduper(script::Module& module) : module_(module) {}
+  void dedup() {
+    for (auto& method : module_.get_methods()) {
+      findModuleUses(method.name());
+    }
+    dedupModuleUses();
+  }
+
+ private:
+  // Analyze the code to record necessary information
+  // to dedup the module uses
+  void findModuleUses(const std::string& method_name) {
+    script::Method method = module_.get_method(method_name);
+    auto graph = method.graph();
+    GRAPH_DUMP("Finding module uses for ", graph);
+
+    std::stack<Block*> blocks_to_visit;
+    blocks_to_visit.push(graph->block());
+    Value* self = graph->inputs()[0];
+    while (!blocks_to_visit.empty()) {
+      Block* b = blocks_to_visit.top();
+      blocks_to_visit.pop();
+      for (Node* n : b->nodes()) {
+        for (Block* subblock : n->blocks()) {
+          blocks_to_visit.push(subblock);
+        }
+        if (n->kind() != prim::CallMethod) {
+          continue;
+        }
+        std::vector<std::string> path;
+        Value* instance = n->inputs()[0];
+        // boundary_val is the value we get when we trace back
+        // the GetAttr access chain until we hit the input of graph
+        // or a node that is not prim::GetAttr
```

Comment:
From the comment it seems that it's possible to find a node that 's neither input nor `prim::GetAttr`. When can this happen? Will all other logic in this pass work correctly in this case?

Reply:
for example when some function call returns a instance, and we call method on that? I think so, if that happens we'll stop the search there and go to the `else` branch starting L802